### PR TITLE
fix(curriculum): combined steps 150 and 151 in js beta RPG

### LINKED ISSUE #54866: js beta build a RPG step 150 and step 151 can be combined
### Describe the Issue

I noticed that step 150 of the RPG js beta curriculum asks us to create an if statement at a specific line in the editor (which happens to be blank) and then in step 151, it tells us to move the line that is currently below the new if statement inside the if block.

I feel this could have been done in one step by just telling us to create an if statement around the line ` monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1;` so we're not having to move it (or move our curly brace around it in a separate step). It seems like something simple to ask and shouldn't require 2 steps to do this?

The current step 150 instruction is:
`In your attack function, below the health variable, create an if statement. Set the condition to call the isMonsterHit function.`

which can become for eg:
`In your attack function, find the line of code that updates the monsterHealth variable and place it within an if block with a condition that calls the isMonsterHit function.`

### Affected Page

https://www.freecodecamp.org/learn/javascript-algorithms-and-data-structures-v8/learn-basic-javascript-by-building-a-role-playing-game/step-150

### Your code

N/A

### Expected behavior

ask the learner to create the if statement around the code instead of create the if in step 150 then move the code into it in step 151

### Screenshots

_No response_

### System

N/A

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by AndrewYturaldi**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #54866

Combined steps 150 and 151 in the javascript beta role player game by removing step 150 and modifying the new step 150. I used the 'Delete Step' command. Tested locally on my machine.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-Move your `monsterHealth` assignment into your `if` block.
+In your `attack` function, find the line of code that updates the `monsterHealth` variable and place it within a new `if` block with a condition that calls the `isMonsterHit` function.
 
 # --hints--
 
-You should move your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` into your `if` block.
+You should move your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` into a new `if` block.
 
 ```js
 assert.match(attack.toString(), /if\s*\(\s*isMonsterHit\(\s*\)\s*\)\s*\{\s*monsterHealth\s*-=\s*weapons\s*\[\s*currentWeapon\s*\]\s*\.power\s*\+\s*Math\.floor\(\s*Math\.random\(\s*\)\s*\*\s*xp\s*\)\s*\+\s*1/)
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md:17]: The tests here will need to be more granular - something like:

- You should have a new `if` block
- Your `if` condition should call the `isMonsterHit` function
- Your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` should be executed when `isMonsterHit()` is true.

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md (hunk 1 above)
Comment (AndrewYturaldi) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md:17]: @naomi-lgbt I was unable to add the You should have a new `if` block test due to there being two other if blocks in the attack function already. I have this test instead:

Your `attack` function should have a new `if` block that calls the `isMonsterHit` function.

I also am not sure how to implement the last test you mentioned:

Your `monsterHealth -= weapons[currentWeapon].power + Math.floor(Math.random() * xp) + 1` should be executed when `isMonsterHit()` is true.

Comment (naomi-lgbt) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-javascript-by-building-a-role-playing-game/62aa204c1e1d33348ff09944.md:17]: >  You should have a new if block test due to there being two other if blocks in the attack function already

Get the number of if blocks with a `String.match(RegExp)` call and assert its length is 3.

**Comment by ojeytonwilliams** [comment]
Sorry for the mix up, @AndrewYturaldi, but the issue has been closed by https://github.com/freeCodeCamp/freeCodeCamp/pull/55095

**Pull request closed without merging**